src/Cli/dotnet/ToolPackage/ToolPackageInstance.cs

```diff
@@ -6,6 +6,7 @@
 using Microsoft.DotNet.Tools;
 using Microsoft.Extensions.EnvironmentAbstractions;
 using NuGet.Frameworks;
+using NuGet.Packaging;
```

Comment:
On `ToolRestoreCommand` line 157, I would lowercase the package.Version going into the Tool Resolver cache. I would also do this in the localToolsResolverCache line 26, 149. I think its possible that data could come to bite us later if we access it somewhere without lowercasing the version.

Also going to ping @dsplaisted to see if he has any thoughts.

Reply:
`ToolRestoreCommand.PackageHasBeenRestored` Loads the Cache of tool restore commands previously executedand then the locations of those tools that were previously restored -> `TryGetMatchingRestoredCommand` -> `LocalTolsResolverCache` `Save` uses `RestoredCommand.Executable` to cache the location of the package, it includes `Version` as well which is not lowercased but that `Version` is not used to generate a path so it's ok, also I think we want to store the true version of the package and not the lowercased one for nuget's directory purposes 
-> `ToolRestoreCommand.InstallPackages` has `IToolPackage` with `Command` with `Executable`  -> `ToolPackageDownloader.InstallPackage` -> Uses Nuget Version in the `Version` but `toolReturnPackageDirectory` for the directory which happens from `GetPackageDirectory` which already lowercased the version.................

TLDR: This is the correct approach and we dont need to lowercase this version I think 😵